rotkehlchen/chain/ethereum/manager.py

```diff
@@ -1342,6 +1344,7 @@ def _update_curve_decoder(self, tx_decoder: 'EVMTransactionDecoder') -> None:
         new_mappings = curve_decoder.reload()
         tx_decoder.address_mappings.update(new_mappings)
 
+    @protect_with_lock()
```

Comment:
If you have a lock here and for example the eth balances try to query also the price for a curve token then you will have one blocked by the other. I was thinking in something like a flag but that won't work either. 

I believe this is the best solution

Reply:
there is no way other than a lock if you want to protect something from simultaneous access if that something can be accessed arbitrarily by a user (in this case by an api caller)